google-cloud-logging/lib/google/cloud/logging/async_writer.rb

```diff
@@ -54,58 +56,37 @@ module Logging
       #                       labels: labels
       #
       class AsyncWriter
-        include Stackdriver::Core::AsyncActor
-
-        DEFAULT_MAX_QUEUE_SIZE = 10000
-        CLEANUP_TIMEOUT = Stackdriver::Core::AsyncActor::CLEANUP_TIMEOUT
-        WAIT_INTERVAL = Stackdriver::Core::AsyncActor::WAIT_INTERVAL
+        include MonitorMixin
 
         ##
-        # @private Item in the log entries queue.
-        QueueItem = Struct.new(:entries, :log_name, :resource, :labels) do
-          def try_combine next_item
-            if log_name == next_item.log_name &&
-               resource == next_item.resource &&
-               labels == next_item.labels
-              entries.concat(next_item.entries)
-              true
-            else
-              false
-            end
-          end
-        end
+        # @private Implementation accessors
+        attr_reader :logging, :max_bytes, :max_count, :interval,
+                    :threads, :max_queue, :partial_success
 
         ##
-        # @private The logging object.
-        attr_accessor :logging
+        # @private Creates a new AsyncWriter instance.
+        def initialize logging, max_count: 10000, max_bytes: 10000000,
+                       max_queue: 100, interval: 5, threads: 10,
+                       partial_success: false
+          @logging = logging
 
-        ##
-        # @private The maximum size of the entries queue, or nil if not set.
-        attr_accessor :max_queue_size
+          @max_count = max_count
+          @max_bytes = max_bytes
+          @max_queue = max_queue
+          @interval  = interval
+          @threads   = threads
 
-        ##
-        # The current state. Either :running, :suspended, :stopping, or :stopped
-        #
-        # DEPRECATED. Use #async_state instead.
-        alias state async_state
+          @partial_success = partial_success
 
-        ##
-        # The last exception thrown by the background thread, or nil if nothing
-        # has been thrown.
-        attr_reader :last_exception
+          @error_callbacks = []
 
-        ##
-        # @private Creates a new AsyncWriter instance.
-        def initialize logging, max_queue_size = DEFAULT_MAX_QUEUE_SIZE,
-                       partial_success = false
-          super()
+          @cond = new_cond
 
-          @logging = logging
-          @max_queue_size = max_queue_size
-          @partial_success = partial_success
-          @queue_resource = new_cond
-          @queue = []
-          @queue_size = 0
+          # Make sure all buffered messages are send when process exits.
```

Comment:
"are send" -> "are sent" ?

Reply:
Fixed. PTAL and resolve this conversation if satisfied.